Modules/Office365DSC/DSCResources/MSFT_SPOStorageEntity/MSFT_SPOStorageEntity.psm1

```diff
@@ -0,0 +1,231 @@
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Key,
+
+        [Parameter()]
+        [System.String]
+        $Value,
+
+        [Parameter()]
+        [ValidateSet("Tenant", "Site")]
+        [System.String]
+        $EntityScope,
+
+        [Parameter()]
+        [System.String]
+        $Comment,
+
+        [Parameter()]
+        [System.String]
+        $Description,
+
+        [Parameter()]
+        [ValidateSet("Present", "Absent")]
+        [System.String]
+        $Ensure = "Present",
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $SiteUrl,
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    Test-PnPOnlineConnection -SiteUrl $SiteUrl -GlobalAdminAccount $GlobalAdminAccount
+
+    $nullReturn = @{
+        Key                = $Key
+        Value              = $Value
+        EntityScope        = $EntityScope
+        Description        = $Description
+        Comment            = $Comment
+        Ensure             = "Absent"
+        SiteUrl            = $SiteUrl
+        GlobalAdminAccount = $GlobalAdminAccount
+    }
+
+    Write-Verbose -Message "Getting storage entity $Key"
+    $CurrentParameters = $PSBoundParameters
+    $CurrentParameters.Clear()
```

Comment:
You should just create a new hashtable instead of grabbing the existing one and removing all items.